src/main/java/net/openhft/hashing/LongTupleHashFunction.java

```diff
@@ -13,55 +13,43 @@
 import static net.openhft.hashing.Util.*;
 
 /**
- * Tuple hash function producing {@code long[]} result from byte sequences of any length and
- * a plenty of different sources which "feels like byte sequences". Except {@link
- * #hashBytes(byte[])}, {@link #hashBytes(ByteBuffer)} (with their "sliced" versions) and
- * {@link #hashMemory(long, long)} methods, which actually accept byte sequences, notion of byte
- * sequence is defined as follows:
+ * Tuple hash function producing more than 64-bit hash code into a result array of type
+ * {@code long[]} from any byte sequences. See {@link LongHashFunction} for the definition of byte
+ * sequence semantics and {@link ByteOrder} requirements for implementations.
+ *
+ * Every {#link LongHashFunction} hash method has two corresponding ones in this class for different
+ * allocation stradges:
  * <ul>
- *     <li>For methods accepting arrays of Java primitives, {@code String}s and
- *     {@code StringBuilder}s, byte sequence is how the input's bytes are actually lay in memory.
+ *     <li>{@code void hash(input..., long[] result)} will store the hash results in array
+ *     {@code result[0 .. newResultArray().length-1]}, and throws exceptions when {@code result}
+ *     is not valid. {@link #newResultArray} method should always be used to create resuable result
```

Comment:
"is not valid" -> "the length of the array is less than newResultArray().length"